src/net/virtual_connection.rs

```diff
@@ -54,6 +59,28 @@ impl VirtualConnection {
         }
     }
 
+    /// Record that this connection has sent a packet. Returns whether the connection has
+    /// become acknowledged because of this send.
+    pub fn record_send(&mut self) -> bool {
+        let was_est = self.is_established();
+        self.ever_sent = true;
+
+        !was_est && self.is_established()
+    }
+
+    /// Record that this connection has sent a packet. Returns whether the connection has
```

Comment:
```suggestion
    /// Records that this connection has sent a packet. Returns whether the connection has
```

Plural start for function comment is a code style in lamianr.